docs/src/pages/guides/FHIRServerUsersGuide.md

```diff
@@ -955,17 +955,20 @@ Thankfully, the HL7 FHIR specification defines a way to express a dependency bet
+To define a local identifier for a resource, you must specify it in the `Bundle.entry.fullUrl` field of the resource's bundle entry. The HL7 FHIR specification recommends that the local identifier be the persistent identity of the resource if known, otherwise a UUID-type value prefixed with `urn:` as in the preceding example. However, the FHIR server does not require the use of a UUID value. You can use any value you like after the `urn:` prefix as long as it is unique within the bundle. For example, you can use `urn:Patient_1`, `urn:ABC`, or `urn:Foo`. Just make sure that no two request entries share the same `fullUrl` value within the request bundle.
 
 After you define a local identifier for the referenced resource, you can then define one or more references to that resource by using the local identifier instead of an external identifier. In the preceding example, you can see that the Observation's `subject.reference` field specifies the Patient's local identifier as specified in the `fullUrl` field of the Patient's request entry.
 
 ### 4.8.1 Processing rules
 The following processing rules apply for the use of local references within a request bundle:
-1.  A local identifier must be defined via a request entry's `fullUrl` field in order for that local identifier to be used in a local reference.
-2.  Local references will only be recognized for local identifiers associated with requst entries with a request method of `POST` or `PUT`.
+1.  A local identifier must be defined via a request bundle entry's `fullUrl` field in order for that local identifier to be used in a local reference.
+2.  Local references will only be recognized for local identifiers associated with request bundle entries with a request method of `POST` or `PUT`.
 3.  `POST` requests will be processed before `PUT` requests.
-4.  There is no order dependency within a request bundle for request entries defining local identifiers, and request entries which reference those local identifiers via local reference. The exception to this rule is for request entries which specify [conditional create](https://www.hl7.org/fhir/http.html#ccreate) or [conditional update](https://www.hl7.org/fhir/http.html#cond-update) requests.
-5.  If a request entry specifying a conditional create or update request defines a local identifier, that request entry must be processed before a request entry which references the local identifier in a local reference.
+4.  There is no order dependency within a request bundle for bundle entries defining local identifiers, and bundle entries which reference those local identifiers via local references. The exception to this rule is for request bundle entries which specify [conditional create](https://www.hl7.org/fhir/http.html#ccreate) or [conditional update](https://www.hl7.org/fhir/http.html#cond-update) requests.
```

Comment:
@punktilious Is the exception in rule 4 and 5 for conditional creates and updates still valid after your re-factoring of the bundle processing in FHIRRestHelper?

Reply:
Good catch - the exception no longer applies because all the ccreate work is now done in a first pass, with the reference resolution in a second pass. The exception sentence in 4. can be removed and all of item 5 can be removed.

Reply:
Should we consider moving this info to the Conformance doc?

Reply:
Done. I considered putting this info in the Conformance doc, but didn't really see a good spot for it.

Reply:
I also rearranged some of the content of section 4.8 so hopefully it's more readable now.

Reply:
thanks